packages/block/src/header.ts

```diff
@@ -569,6 +660,12 @@ export class BlockHeader {
       this.mixHash,
       this.nonce,
     ]
+
+    if (this._common.isActivatedEIP(1559)) {
+      rawItems.push(unpadBuffer(toBuffer(this.baseFeePerGas)))
```

Comment:
since `baseFeePerGas` is already `BN` we don't need to use `toBuffer`, we can do:

```suggestion
      rawItems.push(unpadBuffer(this.baseFeePerGas.toArrayLike(Buffer)))
```

or use the utility function I added to util (we can rename this to `bnToUnpaddedBuffer` as per your issue https://github.com/ethereumjs/ethereumjs-util/issues/284 and maintain an alias from the old name for backwards compat):

```suggestion
      rawItems.push(bnToRlp(this.baseFeePerGas))
```

Reply:
Update: this actually has side effects on instantiation and some tests fail after the change. Will revert locally since I haven't got the time to dig deeper, I think a revert is safe though since this just prevents some other checks in the  main constructor (which would throw) to be reached. Feel free to dig deeper here though.